lib/kb_Metrics/metricsdb_controller.py

```diff
@@ -307,39 +217,34 @@ def get_narratives_from_wsobjs(self, requesting_user, params, token):
                     if wn['name'] == obj['object_name']:
                         wn[u'object_id'] = obj['object_id']
                         wn[u'object_version'] = obj['object_version']
-                        wn[u'latest'] = obj['latest']
                         break
-                elif ':' in wn['name']:
-                    wts = wn['name'].split(':')[1]
-                    p = re.compile(wts, re.IGNORECASE)
-                    if p.search(obj['object_name']):
-                        wn[u'object_id'] = obj['object_id']
-                        wn[u'object_version'] = obj['object_version']
-                        wn[u'latest'] = obj['latest']
+                    elif ':' in wn['name']:
+                        wts = wn['name'].split(':')[1]
+                        if '_' in wts:
+                            wts = wts.split('_')[1]
+                        p = re.compile(wts, re.IGNORECASE)
+                        if p.search(obj['object_name']):
+                            wn[u'object_id'] = obj['object_id']
+                            wn[u'object_version'] = obj['object_version']
                         break
-
         for wn in ws_narrs:
-            if not wn.get('object_id', None) is None:
-                wn[u'last_saved_by'] = wn['username']
-                del wn['username']
-
+            if wn.get('object_id'):
+                wn[u'last_saved_by'] = wn.pop('username')
                 wn[u'nice_name'] = ''
-                if not wn.get('meta', None) is None:
-                    w_meta = wn['meta']
-                    for w_m in w_meta:
-                        if w_m['k'] == 'narrative_nice_name':
-                            wn[u'nice_name'] = w_m['v']
-                    del wn['meta']
+                w_meta = wn['meta']
+                for w_m in w_meta:
+                    if w_m['k'] == 'narrative_nice_name':
+                        wn[u'nice_name'] = w_m['v']
+                        break
+                del wn['meta']
                 ws_narrs1.append(wn)
         return {'metrics_result': ws_narrs1}
 
-    def get_activities_from_wsobjs(self, requesting_user, params, token):
-        if not self.is_admin(requesting_user):
+    def _get_activities_from_wsobjs(self, requesting_user, params, token):
+        if not self._is_metrics_admin(requesting_user):
```

Comment:
do you need to check admin user again since it's only used in _update_daily_activities and metrics_admin is already checked?

Reply:
Now that it is a method, no need to check.